src/promptflow/setup.py

```diff
@@ -45,7 +45,7 @@
     "strictyaml>=1.5.0,<2.0.0",  # used to identify exact location of validation error
     "waitress>=2.1.2,<3.0.0",  # used to serve local service
     "opencensus-ext-azure<2.0.0",  # configure opencensus to send telemetry to azure monitor
-    "ruamel.yaml>=0.17.35,<1.0.0",  # used to generate connection templates with preserved comments
+    "ruamel.yaml==0.17.10",  # used to generate connection templates with preserved comments
```

Comment:
why pin? I think we can use this as lower bound

Reply:
pin to check the error if there are conflict in this list.